src/ASM/DomainDecomposition.C

```diff
@@ -27,109 +27,91 @@
 #include <numeric>
 
 
-//! \brief Iterator for matching nodes on edges/faces with a given orientation and index.
-class NodeIterator {
-public:
-  //! \brief Constructor.
-  //! \param pch Slave patch
-  //! \param orient Orientation of boundary on slave
-  //! \param lIdx Index of boundary on slave
-  //! \param basis Basis to use for boundary on slave
-  //! \param dim Dimension to iterate over
-  NodeIterator(const ASMbase* pch, int orient, int lIdx, int basis, int dim = 2)
-  {
-    if (dim == 0) {
-      nodes.resize(1);
-      return;
-    }
-    const ASMstruct* spch = dynamic_cast<const ASMstruct*>(pch);
-    const ASMunstruct* upch = dynamic_cast<const ASMunstruct*>(pch);
-
-    int nsd = pch->getNoSpaceDim();
-    if (upch) {
-      if (nsd == 3 && dim == 1) {
-        const ASM3D* pch3D = dynamic_cast<const ASM3D*>(pch);
-        nodes = pch3D->getEdge(lIdx, true, basis, -1);
+DomainDecomposition::
+OrientIterator::OrientIterator(const ASMbase* pch,
+                               int orient, int lIdx, int basis, int dim)
+{
+  if (dim == 0) {
+    nodes.resize(1);
```

Comment:
Missing default value? (unitialized, since `int` constructor will not initialize to zero)

Reply:
i guess it has always worked by accident, fixed.